user_sync/rules.py

```diff
@@ -228,14 +230,32 @@ def process_orphaned_dashboard_users(self):
         remove_list_output_path = options['remove_list_output_path']
         remove_nonexistent_users = options['remove_nonexistent_users']
         
+        max_deletions_per_run = options['max_deletions_per_run']
+        max_missing_users = options['max_missing_users']
+
         orphaned_federated_dashboard_users = list(self.iter_orphaned_federated_dashboard_users())
         self.logger.info('Federated orphaned users to be removed: %s', [self.get_dashboard_user_key(dashboard_user) for dashboard_user in orphaned_federated_dashboard_users])        
         
+        number_of_orphaned_dashboard_users = len(orphaned_federated_dashboard_users)
+
         if (remove_list_output_path != None):
             self.logger.info('Writing remove list to: %s', remove_list_output_path)
             self.write_remove_list(remove_list_output_path, orphaned_federated_dashboard_users)
         elif (remove_nonexistent_users):
+            if number_of_orphaned_dashboard_users > max_missing_users:
```

Comment:
Can you make a check for None?  I don't trust how Python handles comparisons would work if max_missing_users is None.